R/textmodel_svmlin.R

```diff
@@ -189,3 +208,107 @@ coefficients.textmodel_svmlin <- function(object, ...) {
 print.predict.textmodel_svmlin <- function(x, ...) {
     print(unclass(x))
 }
+
+
+
+# adapted from RSSL -----------------
+
+svmlin <- function(X, y, X_u = NULL, algorithm = 1, lambda = 1, lambda_u = 1, max_switch = 10000,
+                   pos_frac = 0.5, Cp = 1.0, Cn = 1.0, verbose = FALSE,
+                   intercept = TRUE, scale = FALSE, x_center = FALSE) {
+
+  use_Xu_for_scaling <- TRUE
+  classnames <- levels(y)
+
+    if (scale | x_center) {
+        if (intercept) {
+            cols <- 2:ncol(X) # do not scale the intercept column
+        } else {
+            cols <- 1:ncol(X)
+        }
+
+        if (!is.null(X_u) && use_Xu_for_scaling) {
+            Xe <- rbind(X, X_u)
```

Comment:
If you are not using `X_u` (I don't know that this is for), better to remove from the code. The same applies to other argument.

Reply:
Not using it now, but I read the paper that introduces this method, and there are two semi-supervised methods that use X_u and lambda_u and the switching. It requires unlabelled documents to be added at the time that the SVM is fit to the labelled ones, but I want to understand it better before we offer it. So instead of removing the code, I disabled it so that it could be reactivated later.

When that happens, I will improve and modularise the internal svmlin() call.